### PR TITLE
1.7.1: Use minimum supported Python to build zipapp

### LINKED ISSUE #1514: zipapp for pipx 1.7.0 is missing tomli
**Describe the bug**

When I run the zippapp for 1.7.0 using Python 3.10 I obtain a `ModuleNotFoundError: No module named 'tomli'`.

From this earlier [issue] which was addressed in this [pull request], I understand that the tomli library must be included in the zipapp to support Python < 3.11.  Further I understand that if the zipapp is built using Python 3.11 or later, then tomli will not be included.

The zipapp [artifact](https://github.com/pypa/pipx/actions/runs/10510319885/artifacts/1842776220) uploaded by this CI [job](https://github.com/pypa/pipx/actions/runs/10510319885/job/29118622826) is not affected. Is it possible that the zippapp uploaded to the 1.7.0 release was built using Python 3.11 or later? 

Is it possible to release a new zipapp that includes the tomli library please?

[issue]: https://github.com/pypa/pipx/issues/1142
[pull request]: https://github.com/pypa/pipx/pull/1194/files
[build process]: https://github.com/pypa/pipx/blob/main/.github/workflows/tests.yml

**How to reproduce**

<details>
<summary>Checking zipapp contents</summary>

The easiest way to check the contents is to download the zipapp and extract it:

    curl --location --remote-name https://github.com/pypa/pipx/releases/download/1.7.0/pipx.pyz \
    && unzip pipx.pyz

Then note the absence of `site_packages/tomli`.

For example compare to the previous release:

    curl --location --remote-name https://github.com/pypa/pipx/releases/download/1.6.0/pipx.pyz \
    && unzip pipx.pyz

</details>

<details>
<summary>Reproducing the traceback</summary>

```
Commands to reproduce trace back:

    curl --silent --location --remote-name https://github.com/pypa/pipx/releases/download/1.7.0/pipx.pyz \
    && python3.10 pipx.pyz

Output:

    Traceback (most recent call last):
      File "/home/maxwell-k/Downloads/pipx.pyz/_bootstrap/__init__.py", line 76, in import_string
      File "/home/maxwell-k/.shiv/pipx.pyz_470623b0abb5f46e0cef98e1443fd9cf54f230ecdd5399e59dfd8163372499e1/site-packages/pipx/main.py", line 22, in <module>
        from pipx import commands, constants, paths
      File "/home/maxwell-k/.shiv/pipx.pyz_470623b0abb5f46e0cef98e1443fd9cf54f230ecdd5399e59dfd8163372499e1/site-packages/pipx/commands/__init__.py", line 9, in <module>
        from pipx.commands.run import run
      File "/home/maxwell-k/.shiv/pipx.pyz_470623b0abb5f46e0cef98e1443fd9cf54f230ecdd5399e59dfd8163372499e1/site-packages/pipx/commands/run.py", line 30, in <module>
        import tomli as tomllib
    ModuleNotFoundError: No module named 'tomli'

    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "/home/maxwell-k/Downloads/pipx.pyz/_bootstrap/__init__.py", line 84, in import_string
    AttributeError: module 'pipx' has no attribute 'main'

    During handling of the above exception, another exception occurred:

    Traceback (most recent call last):
      File "/usr/lib64/python3.10/runpy.py", line 196, in _run_module_as_main
        return _run_code(code, main_globals, None,
      File "/usr/lib64/python3.10/runpy.py", line 86, in _run_code
        exec(code, run_globals)
      File "/home/maxwell-k/Downloads/pipx.pyz/__main__.py", line 3, in <module>
      File "/home/maxwell-k/Downloads/pipx.pyz/_bootstrap/__init__.py", line 253, in bootstrap
      File "/home/maxwell-k/Downloads/pipx.pyz/_bootstrap/__init__.py", line 81, in import_string
      File "/home/maxwell-k/Downloads/pipx.pyz/_bootstrap/__init__.py", line 87, in import_string
    ImportError: module 'pipx' has no attribute 'main'
```

</details>

**Expected behavior**

I expect the zipapp to include tomli so that it is functional with Python 3.10.

### PR DESCRIPTION
**Pull request by dukecat0**
Fixes #1514

- [x] I have added a news fragment under `changelog.d/` (if the patch affects the end users)

## Summary of changes
Use minimum supported Python to build zipapp

## Test plan

Tested by running

```
# command(s) to exercise these changes
```

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -98,10 +98,10 @@ jobs:
         uses: actions/checkout@v4
         with:
           ref: "${{ needs.create-tag.outputs.release-tag }}"
-      - name: Set up Python ${{ env.default-python }}
+      - name: Set up Python ${{ env.minimum-supported-python }}
```
Comment (Gitznik) [inline .github/workflows/release.yml:101]: This env variable is not yet defined in this workflow.

**Review by Gitznik — CHANGES_REQUESTED** [review]

**Comment by Gitznik** [comment]
We probably also need a news fragment, so we can create a fix release?

File: .github/workflows/release.yml (hunk 1 above)
Comment (dukecat0) [reply .github/workflows/release.yml:102]: Good catch! Fixed.

**Review by Gitznik — APPROVED** [review]

**Comment by chrysle** [comment]
Merging this PR will trigger a patch release.

**Comment by dukecat0** [comment]
It seems like Github Action has done everything for me in #1517. I will close this now.

**Comment by dukecat0** [comment]
Everything should be good to go now.

**Review by chrysle — APPROVED** [review]

**Pull request merged**